pkgs/applications/science/logic/uhdm/default.nix

```diff
@@ -3,26 +3,41 @@
 , fetchFromGitHub
 , cmake
 , python3
+, gtest
+, fetchpatch
 }:
 
 stdenv.mkDerivation rec {
   pname = "UHDM";
-  version = "0.9.1.40";
+  version = "1.45";
 
   src = fetchFromGitHub {
     owner = "chipsalliance";
     repo = pname;
     rev = "v${version}";
-    hash = "sha256-CliKU2WM8B9012aDcS/mTyIf+JcsVsc4uRRi9+FRWbM=";
+    hash = "sha256-mxQRmI8yUUrSUYa4kUT9URgfqYvuz3V9e1IGjtiHyhc=";
     fetchSubmodules = true;
   };
 
+  patches = [
+    (fetchpatch {
+      url = "https://github.com/chipsalliance/UHDM/commit/ad60fdb65a7c49fdc8ee3fffdca791f9364af4f5.patch";
+      sha256 = "sha256-IkwnepWWmBychJ0mu+kaddUEc9jkldIRq+GyJkhrO8A=";
+      name = "allow-local-gtest.patch";
+    })
+  ];
+
```

Comment:
Why do we need this? When can we remove this again?

A comment would be helpful.

Reply:
This provides the ability to use googletest on the system itself (so that we can use `-DUHDM_USΕ_HOST_GTEST` in the cmake options).
It prevents using the googletest that otherwise comes as submodule of third_party/.

It can be removed with the next tagged version of this project.